src/pineko/kfactor_inclusion.py

```diff
@@ -0,0 +1,291 @@
+            limits.append(
+                (grid.raw.bin_left(dim)[num_bin], grid.raw.bin_right(dim)[num_bin])
+            )
+    norma = grid.raw.bin_normalizations()
+    remap_obj = pineappl.bin.BinRemapper(norma, limits)
+    new_grid.set_remapper(remap_obj)
+    return new_grid
+
+
+def create_grids(
+    gridpath, max_as, first_non_zero_as_order, min_al, centrals_k_factor, alphas
+):
+    """Create all the necessary grids for a certain starting grid."""
+    grid = pineappl.grid.Grid.read(gridpath)
+    m_value = first_non_zero_as_order
+    nec_orders = compute_orders_map(m_value, max_as, min_al)
+    grid_list = {}
+    for to_construct_order in nec_orders:
+        list_grid_order = []
+        for nec_order in nec_orders[to_construct_order]:
+            list_grid_order.append(
+                create_singlegridonly(
+                    grid,
+                    m_value,
+                    nec_order,
+                    to_construct_order,
+                    centrals_k_factor,
+                    alphas,
+                )
+            )
+        grid_list[to_construct_order] = list_grid_order
+
+    return grid_list
+
+
+def write_grids(gridpath, grid_list):
+    """Write the single grids."""
+    base_name = gridpath.stem.split(".pineappl")[0]
+    final_part = ".pineappl.lz4"
+    grid_paths = []
+    for order in grid_list:
+        # For each order, if more than one grid contributes, merge them all together in a single one
+        if len(grid_list[order]) > 1:
+            for grid in grid_list[order][1:]:
+                tmp_path = gridpath.parent / ("tmp" + final_part)
+                grid.raw.write_lz4(tmp_path)
+                grid_list[order][0].raw.merge_from_file(tmp_path)
+                tmp_path.unlink()
+        new_grid_path = gridpath.parent / (
+            base_name + "_new_order" + final_part
+        )  # order[2] is the ren_sv order
+        grid_paths.append(new_grid_path)
+        grid_list[order][0].raw.write_lz4(new_grid_path)
+    return grid_paths
+
+
+def merge_grids(gridpath, grid_list_path, target_path=None):
+    """Merge the single grids in the original."""
+    grid = pineappl.grid.Grid.read(gridpath)
+    base_name = gridpath.stem.split(".pineappl")[0]
+    target_name = base_name + "_pluskfactor.pineappl.lz4"
+    if target_path is None:
+        target_path = gridpath.parent / target_name
+    else:
+        target_path = target_path / target_name
+    for grid_path in grid_list_path:
+        grid.raw.merge_from_file(grid_path)
+        grid_path.unlink()
+    grid.raw.write_lz4(target_path)
+
+
+def is_already_in(to_check, list_orders):
+    """Check if the requested order is already in the grid."""
+    is_in = False
+    for order in list_orders:
+        if order[-2] == 0 and order[-1] == 0:
+            if (order[0] == to_check[0]) and (order[1] == to_check[1]):
+                is_in = True
+                return is_in
+    return is_in
```

Comment:
Why not the short version?
```suggestion
    for order in list_orders:
        if order[-2] == 0 and order[-1] == 0 and (order[0] == to_check[0]) and (order[1] == to_check[1]):
            return True
    return False
```
\+ black of course ;-)